service/src/test/java/bio/terra/cbas/util/methods/TestBackfillOriginalWorkspaceIds.java

```diff
@@ -0,0 +1,289 @@
+        .thenReturn(
+            Arrays.asList(
+                methodVersionClonedNull,
+                methodVersionCreatedNull,
+                methodVersionClonedBackfilled,
+                methodVersionCreatedBackfilled));
+    when(methodVersionDao.updateOriginalWorkspaceId(any(), any())).thenReturn(1);
+
+    BackfillOriginalWorkspaceIds.backfillMethodVersions(
+        methodVersionDao, cbasContextConfig, logger);
+
+    verify(methodVersionDao, times(1)).getMethodVersions();
+    verify(methodVersionDao)
+        .updateOriginalWorkspaceId(methodVersionClonedNull.methodVersionId(), nullWorkspaceId);
+    verify(methodVersionDao)
+        .updateOriginalWorkspaceId(methodVersionCreatedNull.methodVersionId(), currentWorkspaceId);
+    verify(methodVersionDao, never())
+        .updateOriginalWorkspaceId(eq(methodVersionClonedBackfilled.methodVersionId()), any());
+    verify(methodVersionDao, never())
+        .updateOriginalWorkspaceId(eq(methodVersionCreatedBackfilled.methodVersionId()), any());
+  }
+
+  private final OffsetDateTime creationDateBefore =
+      OffsetDateTime.parse("1990-12-19T00:00:00.000000Z");
+  private final OffsetDateTime workspaceCreatedDate =
+      OffsetDateTime.parse("1990-12-20T00:00:00.000000Z");
+  private final OffsetDateTime creationDateAfter =
+      OffsetDateTime.parse("1990-12-21T00:00:00.000000Z");
+
+  private final UUID currentWorkspaceId = UUID.fromString("f760e214-75f8-4afc-abdb-e9945d77a325");
+  private final UUID originalWorkspaceId = UUID.fromString("2f79fc92-ea7d-4465-9a8e-418fdd917c5b");
+  private final UUID nullWorkspaceId = UUID.fromString("00000000-0000-0000-0000-000000000000");
+  private Method methodClonedNull =
+      new Method(
+          UUID.fromString("00000000-0000-0000-0000-000000000010"),
+          "MethodClonedNull",
+          "A method that was cloned from a source workspace, with a null originalWorkspaceId",
+          creationDateBefore,
+          UUID.randomUUID(), // this can be random because it isn't relevant to these tests.
+          "GitHub",
+          null);
+  private Method methodCreatedNull =
+      new Method(
+          UUID.fromString("00000000-0000-0000-0000-000000000020"),
+          "MethodCreatedNull",
+          "A method that was created in the current workspace, with a null originalWorkspaceId",
+          creationDateAfter,
+          UUID.randomUUID(), // this can be random because it isn't relevant to these tests.
+          "GitHub",
+          null);
+
+  private Method methodClonedBackfilled =
+      new Method(
+          UUID.fromString("00000000-0000-0000-0000-000000000030"),
+          "MethodClonedBackfilled",
+          "A method that was cloned from a source workspace, with a backfilled originalWorkspaceId",
+          creationDateBefore,
+          UUID.randomUUID(), // this can be random because it isn't relevant to these tests.
+          "GitHub",
+          originalWorkspaceId);
```

Comment:
If this was cloned and then backfilled, shouldn't this be all-zeros?

Reply:
And for run sets and method versions?

Reply:
whoops, you're right. my intention with these methods/run sets/method versions was to show that the backfiller would skip them if they had any originalWorkspaceId value. I'll change it for clarity, but I don't think it invalidates the tests